app/controllers/concerns/gitis_authentication.rb

```diff
@@ -15,21 +15,34 @@ class InvalidContact < RuntimeError; end
   def current_contact
     return nil unless session[:gitis_contact]
 
-    @current_contact ||= Bookings::Gitis::Contact.new(session[:gitis_contact])
+    @current_contact ||=
+      if Flipper.enabled?(:git_api)
+        GetIntoTeachingApiClient::SchoolsExperienceSignUp.new(session[:gitis_contact])
+      else
+        Bookings::Gitis::Contact.new(session[:gitis_contact])
+      end
   end
 
   def current_contact=(contact)
-    raise InvalidContact unless contact.is_a?(Bookings::Gitis::Contact)
+    valid_classes = [GetIntoTeachingApiClient::SchoolsExperienceSignUp, Bookings::Gitis::Contact]
+    raise InvalidContact unless valid_classes.any? { |klass| contact.is_a?(klass) }
 
-    if current_contact && current_contact.contactid != contact.contactid
-      Rails.logger.warn \
-        "Signed in Candidate overwritten - #{current_contact.contactid} to #{contact.contactid}"
+    current_id = current_contact.try(:contactid) || current_contact.try(:candidate_id)
+    new_id = contact.try(:contactid) || contact.try(:candidate_id)
```

Comment:
The `#try`ing here is a bit tricky to follow, could there be a generic method on the objects here that return the relevant id?

Reply:
Yeah that's a good call; I did it that way as I didn't want to monkey patch the `GetIntoTeachingApiClient::SchoolsExperienceSignUp` but I could instead just modify the existing `Booking::Gitis::Contact` to alias `candidate_id` to `contactid` 👍